.github/workflows/build_publish_lambda_layer.yml

```diff
@@ -22,3 +22,151 @@ jobs:
+          security-group-ids: sg-0fd8d8cd6effda4a5
+          tags: |
+            Name=xuan-cao-ec2-runner
+
+  # build layer on arm64 and amd64, then upload to artifacts
+  build_layer:
+    needs:
+      - start_arm64_runner
+    strategy:
+      fail-fast: false
+      matrix:
+        arch:
+          - x86_64
+          - arm64
+    runs-on: ${{ matrix.arch == 'arm64' && fromJSON(needs.start_arm64_runner.outputs.matrix)['ubuntu:22.04'].label || 'ubuntu-latest' }}
+    steps:
+      - uses: actions/checkout@v3
+
+      # Install additional package for self runner
+      - name: Install additional package for ${{ matrix.arch }}
+        if: ${{ matrix.arch == 'arm64'}}
+        run: |
+          sudo apt-get update
+          echo y | sudo apt-get upgrade
+          echo y | sudo apt-get install python3.10-venv
+
+      - uses: aws-actions/setup-sam@v2
+
+      # Build inside Docker containers
+      # There is only two ruby runtime for aws lambda ruby 3.2 and 2.7
+      # Since the 2.7 is not supported by otel, then we don't use it
+      # BuildArchitecture: arch is for different architecture build
+      - name: build lambda layer
+        run: |
+          uname -a
+          cd lambda/
+          echo "" >> template.yml
+          echo "      BuildArchitecture: ${{ matrix.arch }}" >> template.yml
+          echo "" >> template.yml
+          sam build -u -t template.yml -e BUNDLE_RUBYGEMS__PKG__GITHUB__COM=$GITHUB_RUBY_TOKEN
+          ./zip_layer.sh -n ruby-layer-${{ matrix.arch }}
+        env:
+          GITHUB_RUBY_TOKEN: ${{ secrets.PACKAGE_GITHUB_TOKEN }}
+
+      - name: Upload to artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ruby-layer-${{ matrix.arch }}.zip
+          path: lambda
+
+  # extract the built layer from artifacts, then publish it based on region
+  publish_layer:
+    needs:
+      - build_layer
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        aws_region: 
+          - us-east-1
+        arch:
+          - x86_64
+          - arm64
+    steps:
+      - uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.LAMBDA_PUBLISHER_ARN }}
+          aws-region: ${{ matrix.aws_region }}
+
+      - name: extract layer zip from artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: ruby-layer-${{ matrix.arch }}.zip
+          path: lambda
+
+      - name: extract current solarwinds_apm version
+        run: |
+          APM_VERSION=$(grep "gem 'solarwinds_apm'" lambda/layer/Gemfile | awk -F"'" '{print $4}')
+          APM_VERSION="${APM_VERSION//./_}"
+          echo "SOLARWINDS_APM_VERSION=$APM_VERSION" >> $GITHUB_ENV
+
+      - name: publish lambda layer
+        run: |
+          cd lambda/
+          aws lambda publish-layer-version \
+            --layer-name solarwinds-ruby-layer-$SOLARWINDS_APM_VERSION-${{ matrix.arch }} \
```

Comment:
Let's name our layer `solarwinds-apm-ruby-...` similar to the other language instrumentations [layers](https://swicloud.atlassian.net/wiki/spaces/NIT/pages/3987636225/AWS+Lambda+Instrumentation+Install+Config).

Reply:
Fixed

Reply:
Please remove `-layer`, let's have it just be `solrwinds-apm-ruby-<arch>-<version>`

Reply:
Fixed